contracts/Registry.vy

```diff
@@ -154,13 +174,10 @@ def _get_rates(_pool: address) -> uint256[MAX_COINS]:
             coin: address = self.pool_data[_pool].coins[i]
             if coin == ZERO_ADDRESS:
                 break
-            if coin == self.pool_data[_pool].ul_coins[i]:
+            if coin == self.pool_data[_pool].ul_coins[i] or rate_method_id == 0x00000000:
                 rates[i] = 10 ** 18
             else:
-                rates[i] = convert(
-                    raw_call(coin, rate_method_id, max_outsize=32, is_static_call=True), # dev: bad response
-                    uint256
-                )
+                rates[i] = RateCalc(rate_calc_addr).get_rate(coin, rate_method_id)  # dev: bad response
```

Comment:
Maybe I wasn't clear here.. in the vast majority of cases, the original logic of just calling `rate_method_id` and returning the result will work.  Rather than adding a single always-used rate calculator, we should define it on a per-pool basis.

So it becomes something like:

```python
rate_calc_addr = self.pool_data[_pool].rate_calc_contract
if rate_calc_addr != ZERO_ADDRESS:
    rates[i] = RateCalc(rate_calc_addr).get_rate(coin)
else:
    # the original logic
```